test/e2e/apimachinery/crd_publish_openapi.go

```diff
@@ -336,33 +335,36 @@ func setupCRD(f *framework.Framework, schema []byte, groupSuffix string, version
 	if len(versions) == 0 {
 		return nil, fmt.Errorf("require at least one version for CRD")
 	}
-	apiVersions := []v1beta1.CustomResourceDefinitionVersion{}
-	for _, version := range versions {
-		v := v1beta1.CustomResourceDefinitionVersion{
-			Name:    version,
-			Served:  true,
-			Storage: false,
-		}
-		apiVersions = append(apiVersions, v)
-	}
-	apiVersions[0].Storage = true
 
-	crd, err := crd.CreateMultiVersionTestCRD(f, group, apiVersions, nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create CRD: %v", err)
+	if schema == nil {
```

Comment:
this changed behavior... previously if nil was passed in, we did not set a validation schema on the CRD. the defaulting to `type:object` was only done in the value passed to waitForDefinition